### DISCUSSION TITLE
Replacing packages with busybox functions: Good Idea?

### DISCUSSION THREAD

**Discussion by uwe-007** [termux/termux-packages#11589] [Q&A] [answered]
If I am low on diskspace, can I remove the packages for which a busybox function exists and save diskspace?
Or is this no good idea? Why?

For example:
Packages busybox and wget are installed.
Busybox has wget included.
Is it a good idea to remove the wget package and replace it with link to busybox-wget?
I know, that this will remove all additional files like manpages.

**Comment by sylirre** [comment, marked as answer]
You can, however I don't recommend to touch `coreutils` and `tar` as that may break `dpkg` and `apt`.

Note that you won't free much of disk space. The default installation is about 65-70 MB.
![Screenshot_20220814-203302](https://user-images.githubusercontent.com/107305601/184548297-36099071-48e9-4d8c-9993-1f004ebff2ce.png)

Here is what being pulled into standard installation (dependencies not shown):
```
bash
bzip2
command-not-found
coreutils
curl
dash
debianutils
diffutils
dos2unix
ed
findutils
gawk
grep
gzip
inetutils
less
lsof
nano
net-tools
patch
procps
psmisc
sed
tar
termux-exec
termux-keyring
termux-tools
unzip
util-linux
xz-utils
```

Determine why you need to use specifically Termux. Maybe ConnectBot will be better if you are limited by disk space? Attempting to reduce disk space usage by using Busybox while having extra packages like Python and Clang is counterproductive solution to your issue.

If you need to use Termux but without packages, try this to disable bootstrapping of $PREFIX:
```
# creates custom login script
printf '#!/system/bin/sh\nexport PATH=/system/bin\nexec /system/bin/sh -l\n' > $PREFIX/bin/login

# delete everything in $PREFIX except the login script
find $PREFIX ! -path $PREFIX/bin/login -delete
```
Restart application. This will switch Termux to use only system tools, $PREFIX will consist only of login script.

***

> Packages busybox and wget are installed.

`wget` is not part of default installation.

> Is it a good idea to remove the wget package and replace it with link to busybox-wget?

You may start having trouble with downloading from https sites. If that's not an issue, then feel free to do it.

> I know, that this will remove all additional files like manpages.

If you care about disk space, you will want to remove them first and also `info` and `doc` directories in `$PREFIX/share`. In default installation documentation takes about 20 MB. Think how many of docs you actually use.
